src/tabular/grafter/tabular.clj

```diff
@@ -146,18 +148,19 @@
   "Given a dataset and some columns, narrow the dataset to just the
   supplied columns.
 
-  cols are paired off with columns in the data and then a selection is
-  done.  Any cols left over after the pairing are discarded, but if a
-  selected col is not actually in the data an IndexOutOfBoundsException will
-  be thrown.
+  Supplied cols matched with columns in the data and then a selection is done.
+  Any cols left over after the pairing are discarded, but if a selected col
+  is not actually in the data an IndexOutOfBoundsException will be thrown.
 
   This function can safely be used with infinite sequences."
-
   [dataset cols]
   (let [col-names (column-names dataset)
-        matched-columns (->> cols
+        max-cols (count (:column-names dataset))
+        matched-col-positions (->>
+                             (take max-cols cols)
                              (map (partial col-position col-names)))
-        selected-cols (select-indexed (indexed col-names) matched-columns)]
+        valid-positions (filterv #(not= ::not-found %) matched-col-positions)
+        selected-cols (map #(nth col-names %) valid-positions)]
```

Comment:
This `map` line removes the need for `columns` to depend upon `select-indexed` a significantly more complex function.